sklearn/manifold/_graph_spectral_embedding.py

```diff
@@ -0,0 +1,676 @@
+            Computes randomized svd using
+            :func:`sklearn.utils.extmath.randomized_svd`
+        - 'full'
+            Computes full svd using :func:`scipy.linalg.svd`
+            Does not support ``graph`` input of type scipy.sparse.csr_matrix
+        - 'truncated'
+            Computes truncated svd using :func:`scipy.sparse.linalg.svds`
+
+    Returns
+    -------
+    U : array-like, shape (n_samples, n_components)
+        Left singular vectors corresponding to singular values.
+    D : array-like, shape (n_components)
+        Singular values in decreasing order, as a 1d array.
+    V : array-like, shape (n_components, n_samples)
+        Right singular vectors corresponding to singular values.
+    References
+    ----------
+    .. [1] Zhu, M. and Ghodsi, A. (2006).
+        Automatic dimensionality selection from the scree plot via the use of
+        profile likelihood. Computational Statistics & Data Analysis, 51(2),
+        pp.918-930.
+    """
+    # Added in order to pass check estimator, must include words "one sample"
+    if X.shape[0] == 1:
+        msg = "Input data has only one sample (node)"
+        raise ValueError(msg)
+
+    # Deal with svd_solvers
+    if svd_solver not in ["full", "truncated", "randomized"]:
+        msg = "svd_solver must be one of {full, truncated, randomized}."
+        raise ValueError(msg)
+
+    if svd_solver == "full" and isspmatrix_csr(X):
+        msg = "'full' agorithm does not support scipy.sparse.csr_matrix inputs."
+        raise TypeError(msg)
+
+    if n_components is None:
+        elbows, _ = _select_dimension(X, n_elbows=n_elbows, threshold=None)
+        n_components = elbows[-1]
+    elif not isinstance(n_components, int):
+        msg = "n_components must be an integer, not {}.".format(type(n_components))
+        raise ValueError(msg)
+
+    # Check
+    if (svd_solver == "full") & (n_components > min(X.shape)):
+        msg = "n_components must be <= min(X.shape)."
+        raise ValueError(msg)
+    elif svd_solver == "full":
+        U, D, V = scipy.linalg.svd(X)
+        U = U[:, :n_components]
+        D = D[:n_components]
+        V = V[:n_components, :]
+
+    if (svd_solver in ["truncated", "randomized"]) & (n_components >= min(X.shape)):
+        msg = "n_components must be strictly < min(X.shape)."
+        raise ValueError(msg)
+    elif svd_solver == "truncated":
+        U, D, V = sparse.linalg.svds(X, k=n_components)
+        idx = np.argsort(D)[::-1]  # sort in decreasing order
+        D = D[idx]
+        U = U[:, idx]
+        V = V[idx, :]
+    elif svd_solver == "randomized":
+        U, D, V = randomized_svd(X, n_components)
+
+    return U, D, V
+
+def _select_dimension(
+    X, n_components=None, n_elbows=2, threshold=None, return_likelihoods=False
+):
+    """
+    Generates profile likelihood from array based on Zhu and Godsie method.
+    Elbows correspond to the optimal embedding dimension.
+    Parameters
+    ----------
```

Comment:
This suggestions applies to this doc-string's sections and to other docstrings'.

```suggestion
    Elbows correspond to the optimal embedding dimension.
    
    Parameters
    ----------
```